yoyodyne/models/hmm.py

```diff
@@ -0,0 +1,412 @@
+"""Hard monotonic neural HMM classes."""
```

Comment:
In at least this file it looks like you're using tabs. Please standardize on 4 spaces instead. `s/\t/    /g` ought to fix things.

Reply:
Why are we using the four space convention?

Reply:
That makes me sad. Oh well, looks like my code editor auto-converted.